pkg/tfgen/generate_python.go

```diff
@@ -493,13 +495,16 @@ func (g *pythonGenerator) emitResourceType(mod *module, res *resourceType) (stri
 	w.Writefmtln(
 		`            warnings.warn("explicit use of __opts__ is deprecated, use 'opts' instead", DeprecationWarning)`)
 	w.Writefmtln("            opts = __opts__")
-	w.Writefmtln("        if opts and not isinstance(opts, pulumi.ResourceOptions):")
+	w.Writefmtln("        if opts is None:")
+	w.Writefmtln("            opts = pulumi.ResourceOptions()")
+	w.Writefmtln("        if not isinstance(opts, pulumi.ResourceOptions):")
 	w.Writefmtln("            raise TypeError('Expected resource options to be a ResourceOptions instance')")
-	w.Writefmtln("")
-
-	// Now copy all properties to a dictionary, in preparation for passing it to the base function.  Along
-	// the way, we will perform argument validation.
-	w.Writefmtln("        __props__ = dict()")
+	w.Writefmtln("        if opts.version is None:")
+	w.Writefmtln("            opts.version = utilities.get_version()")
+	w.Writefmtln("        if opts.id is None:")
+	w.Writefmtln("            if __props__ is not None:")
+	w.Writefmtln("                raise TypeError(\"[__props__] should only be provided when [opts.id] was not [None].\")")
```

Comment:
We don't use this `[...]` style anywhere else in similar error messages. Also - this is difficult for me to parse - perhaps instead?

```
__props__ is only valid when passed in combination with a valid opts.id to get an existing resource
```

Reply:
great.